sphinx/themes/basic/static/doctools.js

```diff
@@ -291,27 +288,52 @@ var Documentation = {
   },
 
   initOnKeyListeners: function() {
+    // only install a listener if it is really needed
+    if (!DOCUMENTATION_OPTIONS.NAVIGATION_WITH_KEYS &&
+        !DOCUMENTATION_OPTIONS.SEARCH_WITH_KEYS     &&
+        !DOCUMENTATION_OPTIONS.REMOVE_HIGHLIGHT_WITH_KEYS)
+        return;
     $(document).keydown(function(event) {
       var activeElementType = document.activeElement.tagName;
       // don't navigate when in search box, textarea, dropdown or button
       if (activeElementType !== 'TEXTAREA' && activeElementType !== 'INPUT' && activeElementType !== 'SELECT'
-          && activeElementType !== 'BUTTON' && !event.altKey && !event.ctrlKey && !event.metaKey
-          && !event.shiftKey) {
-        switch (event.keyCode) {
-          case 37: // left
-            var prevHref = $('link[rel="prev"]').prop('href');
-            if (prevHref) {
-              window.location.href = prevHref;
+          && activeElementType !== 'BUTTON') {
+        if (!event.altKey && !event.ctrlKey && !event.metaKey && !event.shiftKey) {
+          switch (event.key) {
+            case 'ArrowLeft':
+              if (!DOCUMENTATION_OPTIONS.NAVIGATION_WITH_KEYS)
+                break;
+              var prevHref = $('link[rel="prev"]').prop('href');
+              if (prevHref) {
+                window.location.href = prevHref;
+                return false;
+              }
+              break;
+            case 'ArrowRight':
+              if (!DOCUMENTATION_OPTIONS.NAVIGATION_WITH_KEYS)
+                break;
+              var nextHref = $('link[rel="next"]').prop('href');
+              if (nextHref) {
+                window.location.href = nextHref;
+                return false;
+              }
+              break;
+            case 'Escape':
+              if (!DOCUMENTATION_OPTIONS.REMOVE_HIGHLIGHT_WITH_KEYS)
+                break;
+              Documentation.hideSearchWords();
               return false;
-            }
-            break;
-          case 39: // right
-            var nextHref = $('link[rel="next"]').prop('href');
-            if (nextHref) {
-              window.location.href = nextHref;
+          }
+        }
+        if (!event.altKey && !event.ctrlKey && !event.metaKey) {
+          // some keyboard layouts need Shift to get /
+          switch (event.key) {
+            case '/':
+              if (!DOCUMENTATION_OPTIONS.SEARCH_WITH_KEYS)
+                break;
+              $('input[name=q]').first().focus();
```

Comment:
I'm not sure if this would work reliably, but if this line were factored out into a function (similar to `hideSearchWords()` above?), HTML themes could probably overwrite this function if they need different behavior?

Reply:
> I'm not sure if it's necessary to have two separate options for `/` and `Esc`. What do you think?

Yes, I agree that one option would be better.

> 
> And naming is of course hard. The name `search_with_keys` is a bit weird, because to search something, you'll have to type the search term, which you typically do with keys, right? So "search" is always "search with keys", isn't it?
> 
> What about using `search_shortcuts` or `enable_search_shortcuts` or something like that? Those would also allow to combine `/` and `Esc`, which are both shortcuts related to search.
> 
> Another (minor) problem with both `search_with_keys` and `remove_highlight_with_keys` is the use of the plural "keys", but it's only a single key in each case (except if `Shift` is needed to get `/`). In the future, more keys might be added, so using the singular "key" also won't be a good idea.

All fixed as I chose `enable_search_shortcuts` name now.

> 
> I've mentioned above ([#9337 (comment)](https://github.com/sphinx-doc/sphinx/pull/9337#issuecomment-899069664)) that this feature is broken for me ... now it works! I guess this is because I'm also using a keyboard where `Shift` has to be pressed to get `/`. Thanks for finding and fixing this @jakobandersen!

Now it should work both on layouts where one needs Shift and one doesn't.

> 
> Finally, it might be interesting to check if this feature plays well with https://github.com/readthedocs/readthedocs-sphinx-search.

Can you please test it for me?

Anyway, thanks for useful comments.

Reply:
Done, function `focusSearchBar` was added.